Formula/openssl110.rb

```diff
@@ -0,0 +1,184 @@
+  if build.with? "test"
+    depends_on :perl => "5.14"
+  else
+    depends_on :perl => "5.10"
+  end
+
+  # This isn't technically needed where Clang exists, but is a small
+  # dep and is hard to reliably express non-Clang-only in dependency format.
+  depends_on "makedepend" => :build
+
+  def arch_args
+    {
+      :x86_64 => %w[darwin64-x86_64-cc enable-ec_nistp_64_gcc_128],
+      :i386   => %w[darwin-i386-cc],
+    }
+  end
+
+  # SSLv2 died with 1.1.0, so no-ssl2 no longer required.
+  # SSLv3 & zlib are off by default with 1.1.0 but this may not
+  # be obvious to everyone, so explicitly state it for now to
+  # help debug inevitable breakage.
+  def configure_args; %W[
+    --prefix=#{prefix}
+    --openssldir=#{openssldir}
+    no-ssl3
+    no-ssl3-method
+    no-zlib
+    shared
+    enable-cms
+  ]
+  end
+
+  def install
+    if build.universal?
+      ENV.permit_arch_flags
+      archs = Hardware::CPU.universal_archs
+    elsif MacOS.prefer_64_bit?
+      archs = [Hardware::CPU.arch_64_bit]
+    else
+      archs = [Hardware::CPU.arch_32_bit]
+    end
+
+    dirs = []
+
+    archs.each do |arch|
+      if build.universal?
+        dir = "build-#{arch}"
+        dirs << dir
+        mkdir dir
+        mkdir "#{dir}/engines"
+      end
+
+      ENV.deparallelize
+      system "perl", "./Configure", *(configure_args + arch_args[arch])
+      system "make", "clean" if build.universal?
+      system "make", "depend"
+      system "make"
+      system "make", "test" if build.with?("test")
+
+      if build.universal?
+        cp "include/openssl/opensslconf.h", dir
+        cp Dir["*.?.?.dylib", "*.a", "apps/openssl"], dir
+        cp Dir["engines/**/*.dylib"], "#{dir}/engines"
+      end
+    end
+
+    system "make", "install", "MANDIR=#{man}", "MANSUFFIX=ssl"
+
+    if build.universal?
+      %w[libcrypto libssl].each do |libname|
+        system "lipo", "-create", "#{dirs.first}/#{libname}.1.1.dylib",
+                                  "#{dirs.last}/#{libname}.1.1.dylib",
+                       "-output", "#{lib}/#{libname}.1.1.dylib"
+        system "lipo", "-create", "#{dirs.first}/#{libname}.a",
+                                  "#{dirs.last}/#{libname}.a",
+                       "-output", "#{lib}/#{libname}.a"
+      end
+
+      Dir.glob("#{dirs.first}/engines/*.dylib") do |engine|
+        libname = File.basename(engine)
+        system "lipo", "-create", "#{dirs.first}/engines/#{libname}",
+                                  "#{dirs.last}/engines/#{libname}",
+                       "-output", "#{lib}/engines/#{libname}"
+      end
+
+      system "lipo", "-create", "#{dirs.first}/openssl",
+                                "#{dirs.last}/openssl",
+                     "-output", "#{bin}/openssl"
+
+      confs = archs.map do |arch|
+        <<-EOS.undent
+          #ifdef __#{arch}__
+          #{(buildpath/"build-#{arch}/opensslconf.h").read}
+          #endif
+          EOS
```

Comment:
Yeah, it is 👍. I'm not sure who put it there originally, but I missed that alignment issue when I re-used it here. Will fix.